.github/workflows/testing.yml

```diff
@@ -0,0 +1,91 @@
+name: Tests
+on:
+  push:
+    branches: [ 8.x-4.x ]
+  pull_request:
+    branches: [ 8.x-4.x ]
+jobs:
+  drupal:
+    name: Drupal ${{ matrix.drupal-core }} (PHP ${{ matrix.php-versions }})
+    runs-on: ubuntu-latest
+    services:
+      mysql:
+        image: mysql:5.7
+        env:
+          MYSQL_ALLOW_EMPTY_PASSWORD: false
+          MYSQL_ROOT_PASSWORD: graphql
+          MYSQL_DATABASE: graphql
+        ports:
+          - 3306/tcp
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+    strategy:
+      fail-fast: false
+      matrix:
+        php-versions: ['7.3', '7.4']
+        drupal-core: ['9.0.x']
+        # Extra run to also test on latest Drupal 8 and PHP 7.2.
+        include:
+          - php-versions: '7.2'
+            drupal-core: '8.9.x'
+    steps:
+      - name: Checkout Drupal core
+        uses: actions/checkout@v2
+        with:
+          repository: drupal/drupal
+          ref: ${{ matrix.drupal-core }}
+      - name: Checkout graphql module
+        uses: actions/checkout@v2
+        with:
+          path: modules/graphql
+      - name: Setup PHP, with composer and extensions
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-versions }}
+          extensions: mbstring, xml, ctype, iconv, intl, pdo_sqlite, mysql
+      - name: Start mysql service
+        run: sudo /etc/init.d/mysql start
+      - name: Get composer cache directory
+        id: composercache
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+      - name: Cache composer dependencies
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.composercache.outputs.dir }}
+          # Use composer.json for key, if composer.lock is not committed.
+          # key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+          restore-keys: ${{ runner.os }}-composer-
```

Comment:
we are restoring cache, but we are not storing it, so restored cache gets always empty?
I would expect to store the cache after composer install/require commands.

Reply:
This is the magic of github actions: this step automatically injects a post-workflow step that does the caching :)

Our caching strategy is not ideal since it uses drupal's composer.json, will switch that to composer.lock where we also install acouple of other dependencies.

Reply:
ah, we already use composer.lock so all good here!

Reply:
wow that's great, making the config more cleaner :)